### PR TITLE
Added NullableTypeForNullDefaultValueSniff

### PR DESCRIPTION
**Pull request by janedbal**
- Enforces using nullability symbol `?` when using `null` as default value, e.g. `bool $param = null` > `?bool $param = null`. 
- I'm not sure with the naming, any better ideas?
- Reason to enforce it: you should use nullability symbol if parameter accepts null value no matter what the default value is. So, if you change from `?bool $param = false` to `?bool $param = null`, you don't need to change typehint - you are changing only default value.

### CODE REVIEW

**Comment by VasekPurchart** [comment]
What about `NullableTypeForNullDefaultValue`? Since this (nullable type) is the name of the feature as per the [RFC](https://wiki.php.net/rfc/nullable_types) (and I have not found any mentions in the docs).

File: SlevomatCodingStandard/Sniffs/TypeHints/NullabilityWhenNullDefaultValueSniff.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+				$ignoreToken1 = array_merge(TokenHelper::$ineffectiveTokenCodes, [T_BITWISE_AND, T_ELLIPSIS]);
+				$typehint = TokenHelper::findPreviousExcluding($phpcsFile, $ignoreToken1, $i - 1, $startPointer);
+
+				if ($typehint === null || !in_array($tokens[$typehint]['code'], [T_NS_SEPARATOR, T_STRING], true)) {
+					continue;
+				};
+
+				$ignoreTokens2 = array_merge(TokenHelper::$ineffectiveTokenCodes, [T_BITWISE_AND, T_ELLIPSIS, T_NS_SEPARATOR, T_STRING]);
+				$beforeTypehint = TokenHelper::findPreviousExcluding($phpcsFile, $ignoreTokens2, $i - 1, $startPointer);
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/TypeHints/NullabilityWhenNullDefaultValueSniff.php]: The third parametr should be `$typehint - 1`. Then you don't need to ignore `T_BITWISE_AND` and `T_ELLIPSIS` here and you can use the same list of type hint tokens from line 51.

**Comment by kukulich** [comment]
@janedbal You can squash your commits. It think one commit should be enough here.

**Comment by janedbal** [comment]
@kukulich Fixed, thank you! 

I'll wait with the squash for your approval so you can validate the changes more easily (divided in commits). Or you can use squash on GitHub :)

**Comment by kukulich** [comment]
@janedbal Superb! You can squash it now. Please don't forget to change commit message so there's the new sniff name :)

**Comment by janedbal** [comment]
@kukulich Squashed

**Pull request merged**